### PR TITLE
Start work on fleet system

### PR DESCRIPTION
**Pull request by quartuxz**
Fleet layoutl, basic hierarchy, fleets, and selection. The fleets are not yet interactable and the scripts for making ships are broken and removed for now.

### CODE REVIEW

File: src/common/components/bodies.cpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include "bodies.h"
+
+cqsp::common::components::bodies::Body::Body() {}
+
+cqsp::common::components::bodies::Body::Body(entt::entity _star_system,
+                                             types::kilometer _radius)
+    : star_system(_star_system), radius(_radius) {}
```
Comment (EhWhoAmI) [inline src/common/components/bodies.cpp:7]: New Line at the end of the file

File: src/common/components/bodies.cpp [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#include "bodies.h"
```
Comment (EhWhoAmI) [inline src/common/components/bodies.cpp:1]: Add copyright, add change this to 
`#include "common/components/bodies.h"` so that we can reduce ambiguity if we ever have another file called bodies.h

File: src/common/systems/navy/sysnavy.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#include "sysnavy.h"
```
Comment (EhWhoAmI) [inline src/common/systems/navy/sysnavy.cpp:1]: Copyright header, include root directory

File: src/common/systems/sysuniversegenerator.cpp [hunk 4]
```diff
@@ -71,4 +85,25 @@ void cqsp::common::systems::universegenerator::ScriptUniverseGenerator::Generate
...
+    cqsp::common::systems::actions::CreateShip(
+        universe, playerFleet,
+        universe.get<cqspc::bodies::Body>(starting_planet).star_system,
+        starting_planet,
+        "pioneer");
+      cqsp::common::systems::actions::CreateShip(
+        universe, playerSubFleet,
+        universe.get<cqspc::bodies::Body>(starting_planet).star_system,
+        starting_planet,
+        "pioneer2");
```
Comment (EhWhoAmI) [inline src/common/systems/sysuniversegenerator.cpp:107]: Ships can be created by scripts, make this a script.

File: src/client/scenes/universeloadingscene.cpp [hunk 5]
```diff
@@ -104,7 +109,7 @@ void cqsp::scene::UniverseLoadingScene::LoadUniverse() {
     script_interface.RegisterDataGroup("events");
 
     using cqsp::asset::TextAsset;
-    // Process scripts for core
+    // Process scripts for cores(loading the scripts?)
```
Comment (EhWhoAmI) [inline src/client/scenes/universeloadingscene.cpp:112]: Remove all the changes in this file, you didn't really change anything code wise, so just revert his file so that we don't have so many files to review.

File: src/common/components/bodies.h [hunk 6]
```diff
@@ -35,6 +35,8 @@ struct Body {
     types::kilometer radius;
 
     entt::entity star_system;
+    Body();
+    Body(entt::entity _star_system, types::kilometer _radius = 0);
```
Comment (EhWhoAmI) [inline src/common/components/bodies.h:39]: Not sure what the constructor is for because entt has a constructor already

File: src/common/components/name.cpp [hunk 7]
```diff
@@ -0,0 +1,6 @@
+#include "name.h"
+
+cqsp::common::components::Name::Name() {}
+
+cqsp::common::components::Name::Name(const std::string &p_name)
+    : name(p_name) {}
```
Comment (EhWhoAmI) [inline src/common/components/name.cpp:6]: Put all of this into the header if you *have* to add this, it's not much code, and doesn't justify adding a new file

Also I don't think we need a constructor because entt already does this for us.

File: src/common/components/player.cpp [hunk 8]
```diff
@@ -0,0 +1 @@
+#include "player.h"
```
Comment (EhWhoAmI) [inline src/common/components/player.cpp:1]: Just delete this file.

File: src/common/components/player.h [hunk 9]
```diff
@@ -16,10 +16,14 @@
...
+#include <entt/entt.hpp>
+
 namespace cqsp {
 namespace common {
 namespace components {
-class Player {};
+struct Player {
+
+};
 }  // namespace components
```
Comment (EhWhoAmI) [inline src/common/components/player.h:27]: Delete all the changes in this file, unnecessary changes

File: src/common/simulation.cpp [hunk 10]
```diff
@@ -35,13 +35,15 @@
 #include "common/systems/economy/sysagent.h"
 #include "common/systems/economy/sysfactory.h"
 #include "common/systems/history/sysmarkethistory.h"
+#include "common/systems//navy/sysnavy.h"
```
Comment (EhWhoAmI) [inline src/common/simulation.cpp:38]: Double forward slash, remove the forward slash, it can be problematic

**Review by EhWhoAmI — CHANGES_REQUESTED** [review]

File: src/common/systems/sysuniversegenerator.cpp (hunk 4 above)
Comment (quartuxz) [reply src/common/systems/sysuniversegenerator.cpp:107]: its a bit hard to do that, I have it planned though

File: src/client/scenes/universeloadingscene.cpp (hunk 5 above)
Comment (quartuxz) [reply src/client/scenes/universeloadingscene.cpp:112]: I think that comment was really unclear sorry

File: src/common/systems/sysuniversegenerator.cpp [hunk 11]
```diff
@@ -49,13 +50,26 @@ void cqsp::common::systems::universegenerator::ScriptUniverseGenerator::Generate
...
+    auto playerFleet = universe.create();
+    universe.emplace<cqspc::Name>(playerFleet, "navy");
+    universe.emplace<cqspc::ships::Fleet>(playerFleet, player);
+    universe.get<cqspc::Civilization>(player).topLevelFleet = playerFleet;
+
+    auto playerSubFleet = universe.create();
+    universe.emplace<cqspc::Name>(playerSubFleet, "vice-navy");
+    universe.emplace<cqspc::ships::Fleet>(playerSubFleet, playerFleet, player, 1);
+    universe.get<cqsps::Fleet>(universe.get<cqspc::Civilization>(player).topLevelFleet).subFleets.push_back(playerSubFleet);
+
```
[the remark this answers is not in the document]
Comment (quartuxz) [reply src/common/systems/sysuniversegenerator.cpp:72]: yes, im working on it.

File: src/common/components/player.h (hunk 9 above)
Comment (quartuxz) [reply src/common/components/player.h:27]: I disagree, its a struct not a class

File: src/common/components/bodies.cpp (hunk 2 above)
Comment (quartuxz) [reply src/common/components/bodies.cpp:1]: deleted it

File: src/common/simulation.cpp [hunk 12]
```diff
@@ -52,9 +54,12 @@ cqsp::common::systems::simulation::Simulation::Simulation(
     AddSystem<cqspcs::SysScript>();
     AddSystem<cqspcs::SysWalletReset>();
 
+    AddSystem<cqspcs::SysNavyControl>();
+
     AddSystem<cqspcs::SysPopulationConsumption>();
     AddSystem<cqspcs::SysMine>();
     AddSystem<cqspcs::SysAgent>();
+    AddSystem<cqspcs::SysNavyControl>();
```
[the remark this answers is not in the document]
Comment (quartuxz) [reply src/common/simulation.cpp:62]: did not.

File: src/common/systems/actions/shiplaunchaction.h [hunk 13]
```diff
@@ -25,10 +25,27 @@ namespace cqsp {
...
 entt::entity CreateShip(cqsp::common::Universe &universe, entt::entity civ,
                         entt::entity orbit, entt::entity starsystem);
+/**
+ *@brief creates a ship with a name and assigns it to a fleet and solar system
+ *@param universe The registry for the entities
+ *@param fleetEnt The fleet to be used
+ *@param starsystemEnt The star system where the ship is created in
+ *@return The ship that is created
+ */
+entt::entity CreateShip(cqsp::common::Universe &universe, entt::entity fleetEnt,
```
[the remark this answers is not in the document]
Comment (quartuxz) [reply src/common/systems/actions/shiplaunchaction.h:41]: yes, I do need it. Working on making a lua function. The top-level fleet is just all ships owned by that civilization.

File: src/common/components/ships.cpp [hunk 14]
```diff
@@ -0,0 +1,16 @@
+#include "ships.h"
```
[the remark this answers is not in the document]
Comment (EhWhoAmI) [reply src/common/components/ships.cpp:1]: Didn't fix this, switch to "common/components/ships.h"

File: src/common/components/ships.cpp [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+    : parentFleet(_parentFleet),
+      owner(_owner),
+      echelon(_echelon){
+
+}
+
+
+cqsp::common::components::ships::Fleet::Fleet(
+    entt::entity _owner)
+    : Fleet(entt::null, _owner, 0) {}
```
Comment (EhWhoAmI) [inline src/common/components/ships.cpp:16]: New line at end of file

File: src/common/components/ships.cpp (hunk 14 above)
[the remark this answers is not in the document]
Comment (EhWhoAmI) [reply src/common/components/ships.cpp:1]: Also copyright

File: src/client/scenes/universeloadingscene.cpp (hunk 5 above)
Comment (EhWhoAmI) [reply src/client/scenes/universeloadingscene.cpp:112]: Can you revert all the changes in this file, none of the changes here were necessary

**Review by EhWhoAmI — COMMENTED** [review]
Still needs a few fixes, but a lot better

File: src/client/scenes/universeloadingscene.cpp (hunk 5 above)
Comment (quartuxz) [reply src/client/scenes/universeloadingscene.cpp:112]: sorry, but that comment was too ambigous
